tests/collections/tts/parts/preprocessing/test_features.py

```diff
@@ -0,0 +1,168 @@
+# Copyright (c) 2023, NVIDIA CORPORATION & AFFILIATES.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import contextlib
+import tempfile
+from pathlib import Path
+
+import numpy as np
+import pytest
+import soundfile as sf
+import torch
+
+from nemo.collections.tts.parts.preprocessing.features import (
+    EnergyFeaturizer,
+    MelSpectrogramFeaturizer,
+    PitchFeaturizer,
+)
+
+
+class TestTTSFeatures:
+    @classmethod
+    def setup_class(cls):
+        cls.audio_filename = "test.wav"
+        cls.spec_dim = 80
+        cls.hop_len = 100
+        cls.audio_len = 10000
+        cls.sample_rate = 20000
+        cls.spec_len = 1 + (cls.audio_len // cls.hop_len)
+        cls.manifest_entry = {"audio_filepath": cls.audio_filename}
+
+    @contextlib.contextmanager
+    def _create_test_dir(self):
+        test_audio = np.random.uniform(size=[self.audio_len])
+        temp_dir = tempfile.TemporaryDirectory()
+        try:
+            test_dir = Path(temp_dir.name)
+            audio_path = test_dir / self.audio_filename
+            sf.write(audio_path, test_audio, self.sample_rate)
+            yield test_dir
+        finally:
+            temp_dir.cleanup()
+
+    @pytest.mark.run_only_on('CPU')
+    @pytest.mark.unit
+    def test_compute_mel_spectrogram(self):
+        mel_featurizer = MelSpectrogramFeaturizer(
+            mel_dim=self.spec_dim, hop_length=self.hop_len, sample_rate=self.sample_rate
```

Comment:
it seems you mixed usage of `cls` and `self` in your class. It is fine to just use `self` for the function `def setup_class(self)`. 

Besides, when using `cls`, adding a decor `@classmethod` is redundant.